impeller/renderer/command.h

```diff
@@ -28,13 +28,26 @@ namespace impeller {
 template <class T>
 struct Resource {
   using ResourceType = T;
-  const ShaderMetadata* isa;
   ResourceType resource;
 
-  Resource() : isa(nullptr) {}
+  Resource() : isa_(nullptr) {}
 
-  Resource(const ShaderMetadata* p_isa, ResourceType p_resource)
-      : isa(p_isa), resource(p_resource) {}
+  Resource(const ShaderMetadata* isa, ResourceType p_resource)
+      : resource(p_resource), isa_(isa) {}
+
+  Resource(std::shared_ptr<const ShaderMetadata>& isa, ResourceType p_resource)
+      : resource(p_resource), isa_(nullptr), dynamic_isa_(isa) {}
+
+  const ShaderMetadata* isa() const {
+    return dynamic_isa_ ? dynamic_isa_.get() : isa_;
+  }
+
+ private:
+  // Static shader metadata (typically generated by ImpellerC).
+  const ShaderMetadata* isa_;
```

Comment:
Default it to `nullptr` here in case we forget to explicitly do so in the ctor now or if more are added.

Reply:
done